sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -146,38 +147,44 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
     makeTablesDataset(plan)
   }
 
-  private def makeTablesDataset(plan: ShowTables): Dataset[Table] = {
-    val qe = sparkSession.sessionState.executePlan(plan)
-    val catalog = qe.analyzed.collectFirst {
-      case ShowTables(r: ResolvedNamespace, _, _) => r.catalog
-      case _: ShowTablesCommand =>
-        sparkSession.sessionState.catalogManager.v2SessionCatalog
-    }.get
-    val tables = qe.toRdd.collect().map { row =>
-      val tableName = row.getString(1)
-      val namespaceName = row.getString(0)
-      val isTemp = row.getBoolean(2)
+  private[sql] def resolveTable(row: InternalRow, catalog: CatalogPlugin): Option[Table] = {
```

Comment:
it seems we only need to pass in the catalog name

Reply:
right